### ISSUE TITLE
Searching in files through Search window hangs indefinitely (and cannot be halted)

### ISSUE THREAD

**Issue by paulmadejong** [microsoft/vscode#201995]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.85.1 (Universal)
- OS Version: macOS Sonoma 14.2.1 (23C71) (M1 Pro)

Steps to Reproduce:

1. Open specific project using workspace file or through File Open Workspace 
2. Type any search item in Search window (with or without files to include/exclude)
3. Search runs indefinitely, with a blue "progress bar" running and no response to ESC or clicking the "Cancel Search" button
4. No search results are obtained
5. Click on "Open New Search Editor"
6. Type in same search item in new editor window
7. Search works fine and returns results in new editor window
8. Use search window/pane again, but this doesn't work as it's still stuck....

Strangely enough, for another project (in a different folder), the search runs fine.

The issues seems related to #201964 but my issue is present on my local filesystem.

**Comment by andreamah** [comment]
Is it a particularly large repo, and what type of files are in the workspace?

**Comment by paulmadejong** [comment]
> Is it a particularly large repo, and what type of files are in the workspace?

Roughly 10800 files. The repo that works had about 5500 files. But as stated before, the "New Search Editor" window works fine, just the regular search doesn't.

**Comment by andreamah** [comment]
Hmm, strange. Has this only been happening recently? Do you get any results while the process is hanging (ie: already getting all the results, but the search panel keeps loading)? Can you provide a profile of the renderer process? https://github.com/microsoft/vscode/wiki/Performance-Issues#profiling-the-renderer-process

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by paulmadejong** [comment]
Sorry, haven't been able to respons sooner. Anyway, the profiling doesn't work, clicking the record button doesn't do anything. Alos, the issue of an indefinite search only happens for a certain (large) workspace. It only finds one result but then keeps on searching without retrieving anything more nor being able to stop and perform another search. As said before, a search in the additional search editor pane works normally but not in the search bar. 

I have been having this issue since November. The workspace hasn't increased dramatically in that time and has always been large. I was hoping for a fix in an mostly update but so far it hasn't been fixed and hence me reporting this issue here.

**Comment by andreamah** [comment]
Do you have the exact same toggles and files to include/exclude in your search editor? Can you send a screenshot of the configuration in both?

**Comment by paulmadejong** [comment]
> Do you have the exact same toggles and files to include/exclude in your search editor? Can you send a screenshot of the configuration in both?

Yes I do. But interestingly, I "fixed" the issue myself. When running the profiler, I noticed the Python/Jupyter interactive window that I had open was glitching. I killed it and ran another search, this worked like normal. I then opened a new Interactive Window and search remained working so it seems fixed for me for now.

I still don't know what the Git Error is is wrote above, the error itself is also not helping to solve it.
